shipyard_blueprints/ftp/shipyard_ftp/cli/upload.py

```diff
@@ -41,229 +42,87 @@ def get_args():
     return parser.parse_args()
 
 
-def extract_file_name_from_source_full_path(source_full_path):
-    """
-    Use the file name provided in the source_full_path variable. Should be run
-    only if a destination_file_name is not provided.
-    """
-    destination_file_name = os.path.basename(source_full_path)
-    return destination_file_name
-
-
-def enumerate_destination_file_name(destination_file_name, file_number=1):
-    """
-    Append a number to the end of the provided destination file name.
-    Only used when multiple files are matched to, preventing the destination
-    file from being continuously overwritten.
-    """
-    if re.search(r"\.", destination_file_name):
-        destination_file_name = re.sub(
-            r"\.", f"_{file_number}.", destination_file_name, 1
+def main():
+    try:
+        args = get_args()
+        host = args.host
+        port = args.port
+        username = args.username
+        password = args.password
+        source_file_name = args.source_file_name
+        source_folder_name = args.source_folder_name
+        destination_filename = args.destination_file_name or source_file_name
+        destination_folder_name = shipyard.clean_folder_name(
+            args.destination_folder_name
         )
-    else:
-        destination_file_name = f"{destination_file_name}_{file_number}"
-    return destination_file_name
-
-
-def determine_destination_file_name(
-    *, source_full_path, destination_file_name, file_number=None
-):
-    """
-    Determine if the destination_file_name was provided, or should be extracted
-    from the source_file_name, or should be enumerated for multiple file
-    uploads.
-    """
-    if destination_file_name:
-        if file_number:
-            destination_file_name = enumerate_destination_file_name(
-                destination_file_name, file_number
+        source_file_name_match_type = args.source_file_name_match_type
+        errors = []
+        ftp_client = FtpClient(host=host, port=port, user=username, pwd=password)
+        if source_file_name_match_type == "exact_match":
+            source_full_path = shipyard.combine_folder_and_file_name(
+                folder_name=f"{os.getcwd()}/{source_folder_name}",
```

Comment:
Is it necessary to join the current working directory to the source folder name instead of just provide the source folder name as the argument?

Reply:
I believe that just the source folder should work but the original function had it so I opted for the better save than sorry route